app/controllers/predicates/actions/AuthorisedAction.scala

```diff
@@ -53,6 +53,7 @@ class AuthorisedAction @Inject() (appConfig: AppConfig)(implicit val authService
     authService.authorised().retrieve(affinityGroup) {
       case Some(AffinityGroup.Agent) => agentAuthentication(block)(request, headerCarrier)
       case Some(affinityGroup)       => individualAuthentication(block, affinityGroup)(request, headerCarrier)
+      case _                         => throw new UnauthorizedException("Unable to retrieve affinityGroup")
```

Comment:
@lukastymo What do you think about this change? It's similar to what we do in SE, but not the same as we don't retrieve the internal id.

Reply:
It looks better; I would probably use `case None` instead of _ to reduce thinking overhead while reading